### PR TITLE
FT: adding more server handlers

### PR DESCRIPTION
**Pull request by nicolas2bert**
Add `checkContinue` and `checkExpectation` server events.

### CODE REVIEW

File: lib/network/http/server.js [hunk 1]
```diff
@@ -323,6 +351,34 @@ class Server {
...
+     * Function called when request with an HTTP Expect header is received,
+     * where the value is not 100-continue
+     *
+     * @param {http.IncomingMessage|https.IncomingMessage} req - Request object
+     * @param {http.ServerResponse|https.ServerResponse} res - Response object
+     * @return {undefined}
+     */
+    _onCheckExpectation(req, res) {
+        // Setting no delay of the socket to the value configured
+        req.connection.setNoDelay(this.isNoDelay());
```
Comment (jonathan-gramain) [inline lib/network/http/server.js]: Why it this done here, any link with `Expect` header? Shouldn't it be done in the top-level request handler instead?

**Comment by LaurenSpiegel** [comment]
Thanks for doing this, will you please also add the on connection listener and log like this:
https://github.com/scality/S3/blob/master/lib/server.js#L70

**Comment by nicolas2bert** [comment]
@jonathan-gramain agreed

**Comment by nicolas2bert** [comment]
@ironman-machine ok

**Comment by rahulreddy** [comment]
@nicolas2bert Can you rebase this from master and fix the conflicts?

**Comment by nicolas2bert** [comment]
Done @rahulreddy

**Comment by nicolas2bert** [comment]
@ironman-machine ok

**Pull request merged**